waltz-server/src/test/java/com/wepay/waltz/store/internal/StoreImplTest.java

```diff
@@ -0,0 +1,134 @@
+package com.wepay.waltz.store.internal;
+
+import com.wepay.waltz.common.metadata.PartitionMetadata;
+import com.wepay.waltz.common.metadata.PartitionMetadataSerializer;
+import com.wepay.waltz.common.metadata.ReplicaId;
+import com.wepay.waltz.common.metadata.ReplicaState;
+import com.wepay.waltz.common.metadata.StoreMetadata;
+import com.wepay.waltz.common.metadata.StoreParams;
+import com.wepay.waltz.storage.WaltzStorage;
+import com.wepay.waltz.test.util.IntegrationTestHelper;
+import com.wepay.waltz.test.util.WaltzStorageRunner;
+import com.wepay.zktools.zookeeper.ZNode;
+import com.wepay.zktools.zookeeper.ZooKeeperClient;
+import com.wepay.zktools.zookeeper.internal.ZooKeeperClientImpl;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+public class StoreImplTest {
+    private static final int NUM_REPLICAS = 3;
+    private static final int CLUSTER_NUM_PARTITIONS = 3;
+
+    @Test
+    public void testReplicaAssignmentChangeEffects() throws Exception {
```

Comment:
Looking for suggestions/thoughts on the testing front...

This is more of an integration test than a unit test, please let me know if it should be at a different place...